hoover/search/es.py

```diff
@@ -1,17 +1,34 @@
+import json
+from contextlib import contextmanager
 from django.conf import settings
 from elasticsearch import Elasticsearch
 from elasticsearch.client.utils import _make_path
 from elasticsearch.exceptions import NotFoundError, RequestError, ConnectionError
 
-es = Elasticsearch(settings.HOOVER_ELASTICSEARCH_URL)
 DOCTYPE = 'doc'
 
 class SearchError(Exception):
     def __init__(self, reason):
         self.reason = reason
 
+@contextmanager
+def elasticsearch():
```

Comment:
I can't easily rebase the code to only contain that part of the patch set, because I was factoring out a bunch of stuff while implementing batch search (because there are several common operations, like `_get_indices`.

So I either have o reset everything and go through everything hunk by hunk or rebase 11 commits twice, so I get the context manager without batch search and the batch search based on that context manager.

It would be much less work to accept this PR when it's ready and rebase your work over it. You have a [single commit that touches `es.py`](https://github.com/hoover/search/commit/e28bbe33359ee27ea05fb94244fd567e7bad7f0e), just rebase that one.